OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/ide/instances/lcp_on_fields/problem/LCPOnFieldsEdgeFunctions.scala

```diff
@@ -0,0 +1,510 @@
+
+/**
+ * Edge function holding the current object state (in form of its field-value mapping).
+ *
+ * @author Robin Körkemeier
+ */
+case class ObjectEdgeFunction(
+    values: immutable.Map[String, LinearConstantPropagationValue]
+) extends EdgeFunction[LCPOnFieldsValue] {
+    override def compute(sourceValue: LCPOnFieldsValue): LCPOnFieldsValue =
+        sourceValue match {
+            case UnknownValue         => UnknownValue
+            case ObjectValue(values2) => ObjectValue((values -- values2.keys) ++ values2)
+            case VariableValue        => ObjectValue(values)
+
+            case _ =>
+                throw new UnsupportedOperationException(s"Computing $this for $sourceValue is not implemented!")
+        }
+
+    override def composeWith(secondEdgeFunction: EdgeFunction[LCPOnFieldsValue]): EdgeFunction[LCPOnFieldsValue] =
+        secondEdgeFunction match {
+            case ObjectEdgeFunction(values2) =>
+                ObjectEdgeFunction((values -- values2.keys) ++ values2)
+
+            case PutFieldEdgeFunction(fieldName, value) =>
+                ObjectEdgeFunction((values - fieldName) + (fieldName -> value))
+
+            case IdentityEdgeFunction()   => this
+            case AllTopEdgeFunction(_)    => secondEdgeFunction
+            case AllBottomEdgeFunction(_) => secondEdgeFunction
+
+            case _ =>
+                throw new UnsupportedOperationException(s"Composing $this with $secondEdgeFunction is not implemented!")
+        }
+
+    override def meetWith(otherEdgeFunction: EdgeFunction[LCPOnFieldsValue]): EdgeFunction[LCPOnFieldsValue] =
+        otherEdgeFunction match {
+            case ObjectEdgeFunction(values2) =>
+                ObjectEdgeFunction(
+                    values.keySet
+                        .union(values2.keySet)
+                        .map { fieldName =>
+                            fieldName -> LinearConstantPropagationLattice.meet(
+                                values.getOrElse(fieldName, linear_constant_propagation.problem.VariableValue),
+                                values2.getOrElse(fieldName, linear_constant_propagation.problem.VariableValue)
+                            )
+                        }
+                        .toMap
+                )
+
+            case PutFieldEdgeFunction(fieldName, value) =>
+                ObjectEdgeFunction(
+                    (values - fieldName).map { case (fieldName2, _) =>
+                        fieldName2 -> linear_constant_propagation.problem.VariableValue
+                    } +
+                        (fieldName -> LinearConstantPropagationLattice.meet(
+                            value,
+                            values.getOrElse(fieldName, linear_constant_propagation.problem.VariableValue)
+                        ))
+                )
+
+            case IdentityEdgeFunction()   => this
+            case AllTopEdgeFunction(_)    => this
```

Comment:
If you don't need to match the parameters, you can replace cases like this with, e.g., `case _: AllTopEdgeFunction`